src/core/border_router/routing_manager.hpp

```diff
@@ -97,7 +97,6 @@ class RoutingManager : public InstanceLocator
 {
     friend class ot::Notifier;
     friend class ot::Instance;
-
```

Comment:
I guess you wanted to declare the `BorderAgent` as a `friend` and that why changed this? :)

We can do it the other way, i.e., `RoutingManager` can be defined as a `friend` of `BorderAgent` and then we can make the `PostNotifyMeshCoPServiceChangeTask()`.

The way it is done now in the PR is also fine, keeping `PostNotifyMeshCoPServiceChangeTask()` as `public` method.